fflas-ffpack/fflas/fflas_faxpy.inl

```diff
@@ -27,18 +28,168 @@
+    {
+        Y = SimdT::fmadd(Y,A,X);
+        VEC_MOD(Y, H);
+    }
+
+    template<class Field>
+    inline typename std::enable_if<FFLAS::support_simd_mod<typename Field::Element>::value, void>::type
+    axpyp(const Field &F, const typename Field::Element a, typename Field::ConstElement_ptr X, typename Field::Element_ptr Y, const size_t n, HelperMod<Field> &G)
+    {
+        typedef typename Field::Element Element;
+        using simd = Simd<Element>;
+        using vect_t = typename simd::vect_t;
+        HelperModSimd<Field,simd> H(F,G);
+        vect_t A, U, V;
+        A = simd::set1(a);
+
+        size_t i = 0;
+
+        if (n < simd::vect_size)
+        {
+            for (; i < n ; i++)
+            {
+                Y[i] = reduce(a*X[i]+Y[i], H);
+            }
+            return;
+        }
+
+        long st = long(Y) % simd::alignment;
+        if (st)
+        { // the array Y is not aligned (process few elements until (Y+i) is aligned)
+            for (size_t j = static_cast<size_t>(st) ; j < simd::alignment ; j+=sizeof(Element), i++)
+            {
+                Y[i] = reduce(a*X[i]+Y[i], H);
+            }
+        }
+        FFLASFFPACK_check((long(Y+i) % simd::alignment == 0));
+        if (long(Y+i)%simd::alignment==0)
+        {
+            // perform the loop using SIMD
+            for (;i <= n - simd::vect_size ; i += simd::vect_size)
+            {
+                U = simd::loadu(X+i); // could be transformed into a load at the cost of an additional top check. Worth it?
+                V = simd::load(Y+i);
+                VEC_FMA(A, U, V, H);
+                simd::store(Y+i,V);
+            }
+        }
+        // perform the last elt from T without SIMD
+        for (; i < n ; i++)
+        {
+            Y[i] = reduce(a*X[i]+Y[i], H);
+        }
+    }
+
+#endif // __FFLASFFPACK_HAVE_SSE4_1_INSTRUCTIONS
+
+    template<class Field>
+    inline typename std::enable_if<!FFLAS::support_simd_mod<typename Field::Element>::value && FFLAS::support_fast_mod<typename Field::Element>::value, void>::type
+    axpyp(const Field &F, const typename Field::Element a, typename Field::ConstElement_ptr X, typename Field::Element_ptr Y, const size_t n, HelperMod<Field> &H)
+    {
+        size_t i = 0;
+
+        for (; i < n ; i++)
+        {
+            Y[i] = reduce(a*X[i] + Y[i], H);
+        }
+    }
+
+
+    template<class Field>
+    inline typename std::enable_if<FFLAS::support_fast_mod<typename Field::Element>::value, void>::type
+    axpyp(const Field &F, const typename Field::Element a, typename Field::ConstElement_ptr X, typename Field::Element_ptr Y, const size_t n, const size_t incX, const size_t incY,
+              HelperMod<Field> &H)
+    {
+        typename Field::ConstElement_ptr Xi = X;
+        typename Field::Element_ptr Yi=Y;
+        for (; Xi < X+n*incX; Xi+=incX, Yi+=incY )
+            *Yi = reduce(a*(*Xi) + (*Yi), H);
```

Comment:
Are we sure that the infix operators + and * are legit for this field?

Reply:
This should come from support_fast_mod (also implied by support_simd_mod)

Reply:
Ok. At some point, one should consider cleaning up these traits. This is #304

Reply:
This axpyp should also not be in the namespace vectorised as it does not use vectorization